sys/include/usb/hid.h

```diff
@@ -103,6 +105,375 @@ typedef struct __attribute__((packed)){
     uint16_t report_length; /**< the total size of the Report descriptor. */
 } usb_desc_hid_t;
 
+/**
+ * @defgroup usb_hid_descriptor USB HID descriptor helpers
+ * @ingroup  usb_hid
+ *
+ * @brief USB HID descriptor format helpers
+ * @{
+ */
+
+
+/**
+ * @brief HID Main item type
+ */
+#define USB_HID_ITEM_TYPE_MAIN      0x0
+
+/**
+ * @brief HID Global item type
+ */
+#define USB_HID_ITEM_TYPE_GLOBAL        0x1
+
+/**
+ * @brief HID Local item type
+ */
+#define USB_HID_ITEM_TYPE_LOCAL     0x2
+
+/**
+ * @brief HID Input item tag
+ */
+#define USB_HID_ITEM_TAG_INPUT      0x8
+
+/**
+ * @brief HID Output item tag
+ */
+#define USB_HID_ITEM_TAG_OUTPUT     0x9
+
+/**
+ * @brief HID Collection item tag
+ */
+#define USB_HID_ITEM_TAG_COLLECTION     0xA
+
+/**
+ * @brief HID Feature item tag
+ */
+#define USB_HID_ITEM_TAG_FEATURE        0xB
+
+/**
+ * @brief HID End Collection item tag
+ */
+#define USB_HID_ITEM_TAG_COLLECTION_END 0xC
+
+/**
+ * @brief HID Usage Page item tag
+ */
+#define USB_HID_ITEM_TAG_USAGE_PAGE     0x0
+/**
+ * @brief HID Logical Minimum item tag
+ */
+#define USB_HID_ITEM_TAG_LOGICAL_MIN    0x1
+
+/**
+ * @brief HID Logical Maximum item tag
+ */
+#define USB_HID_ITEM_TAG_LOGICAL_MAX    0x2
+
+/**
+ * @brief HID Physical Minimum item tag
+ */
+#define USB_HID_ITEM_TAG_PHYSICAL_MIN   0x3
+
+/**
+ * @brief HID Physical Maximum item tag
+ */
+#define USB_HID_ITEM_TAG_PHYSICAL_MAX   0x4
+
+/**
+ * @brief HID Unit Exponent item tag
+ */
+#define USB_HID_ITEM_TAG_UNIT_EXPONENT  0x5
+
+/**
+ * @brief HID Unit item tag
+ */
+#define USB_HID_ITEM_TAG_UNIT       0x6
+
+/**
+ * @brief HID Report Size item tag
+ */
+#define USB_HID_ITEM_TAG_REPORT_SIZE    0x7
+
+/**
+ * @brief HID Report ID item tag
+ */
+#define USB_HID_ITEM_TAG_REPORT_ID      0x8
+
+/**
+ * @brief HID Report count item tag
+ */
+#define USB_HID_ITEM_TAG_REPORT_COUNT   0x9
+
+/**
+ * @brief HID Usage item tag
+ */
+#define USB_HID_ITEM_TAG_USAGE      0x0
+
+/**
+ * @brief HID Usage Minimum item tag
+ */
+#define USB_HID_ITEM_TAG_USAGE_MIN      0x1
+
+/**
+ * @brief HID Usage Maximum item tag
+ */
+#define USB_HID_ITEM_TAG_USAGE_MAX      0x2
+
+/**
+ * @brief Physical collection type
+ */
+#define USB_HID_COLLECTION_PHYSICAL     0x00
+
+/**
+ * @brief Application collection type
+ */
+#define USB_HID_COLLECTION_APPLICATION  0x01
```

Comment:
Any reasons not to declare
`USB_HID_COLLECTION_LOGICAL`
`USB_HID_COLLECTION_REPORT`
... and so on ?

Reply:
Nope, added them.